app/cdap/components/Metadata/SearchSummary/EntityTopBar/index.tsx

```diff
@@ -55,16 +57,35 @@ const EntityTab = styled(Tab)`
 
 interface IEntityTopBarProps {
   goBack: () => void;
-  entityType: string;
+  defaultTab: number;
   entityId: string;
+  entityType: string;
+  query: string;
 }
 
-const EntityTopBar: React.FC<IEntityTopBarProps> = ({ goBack, entityType, entityId }) => {
+const EntityTopBar: React.FC<IEntityTopBarProps> = ({
+  goBack,
+  defaultTab,
+  entityType,
+  entityId,
+  query,
+}) => {
   const entityTypeTabAriaLabel = T.translate(`${I18N_PREFIX}.entityTypeTabAriaLabel`);
-  const [entityTypeTab, setEntityTypeTab] = useState(0);
+  const [redirectUrl, setRedirectUrl] = useState(null);
+
+  if (redirectUrl) {
+    return <Redirect to={redirectUrl} />;
+  }
 
-  function handleTabChange(event, newValue: string) {
-    setEntityTypeTab(Number(newValue));
+  function handleTabChange(event, newValue: number) {
+    if (defaultTab !== newValue) {
+      const baseUrl = `/ns/${getCurrentNamespace()}/metadata/${entityType}/${entityId}/`;
```

Comment:
Is there any way to leverage the router so we don't need to put URLs in the component?